packages/office-ui-fabric-react/src/components/Panel/Panel.styles.ts

```diff
@@ -158,102 +230,39 @@ export const getStyles = (props: IPanelStyleProps): IPanelStyles => {
-          '$root &': [
-            isOpen && {
-              pointerEvents: 'auto'
-            },
-            type === PanelType.smallFluid && {
-              width: '100%'
-            },
-            type === PanelType.smallFixedNear && {
-              right: 'auto',
-              left: 0,
-              width: panelSize.width.xs,
-              boxShadow: '0px 0px 30px 0px rgba(0,0,0,0.2)'
-            },
-            type === PanelType.smallFixedFar && {
-              width: panelSize.width.xs,
-              selectors: {
-                ['@media (min-width: ' + ScreenWidthMinMedium + 'px)']: {
-                  width: panelSize.width.sm
-                }
-              }
-            },
-            type === PanelType.medium && {
-              selectors: {
-                ['@media (min-width: ' + ScreenWidthMinUhfMobile + 'px)']: {
-                  left: panelSize.margin.md,
-                  width: 'auto'
-                },
-                ['@media (min-width: ' + ScreenWidthMinXLarge + 'px)']: {
-                  left: 'auto',
-                  width: panelSize.width.md
-                }
-              }
-            },
-            (type === PanelType.large || type === PanelType.largeFixed) && {
-              selectors: {
-                ['@media (min-width: ' + ScreenWidthMinUhfMobile + 'px)']: {
-                  left: panelSize.margin.md,
-                  width: 'auto'
-                },
-                ['@media (min-width: ' + ScreenWidthMinXXLarge + 'px)']: {
-                  left: panelSize.margin.lg
-                }
-              }
-            },
-            type === PanelType.largeFixed && {
-              selectors: {
-                ['@media (min-width: ' + ScreenWidthMinXXLarge + 'px)']: {
-                  left: 'auto',
-                  width: panelSize.width.lg
-                }
-              }
-            },
-            type === PanelType.extraLarge && {
-              selectors: {
-                ['@media (min-width: ' + ScreenWidthMinUhfMobile + 'px)']: {
-                  left: panelSize.margin.md,
-                  width: 'auto'
-                },
-                ['@media (min-width: ' + ScreenWidthMinXXLarge + 'px)']: {
-                  left: panelSize.margin.xl
-                }
-              }
-            },
-            isCustomPanel && {
-              maxWidth: '100vw'
-            }
-          ]
-        }
-      },
-      {
-        display: 'flex',
-        flexDirection: 'column',
         maxHeight: '100%',
+        bottom: 0,
+        top: 0,
+        // (left, right, width) - Properties to be overridden depending on the type of the Panel and the screen breakpoint.
+        left: panelMargin.auto,
+        right: panelMargin.none,
+        width: panelWidth.full,
         selectors: {
           ['@supports (-webkit-overflow-scrolling: touch)']: {
             maxHeight: windowHeight
-          }
+          },
+          ...getPanelBreakpoints(type!)
```

Comment:
Will `type` ever be undefined? If so, how should that be handled? (if it won't be undefined, you should ideally find a way to update the relevant variable types to avoid the `!`)

Reply:
It will never be `undefined` because we are making sure it will always be of at least `PanelType.smallFixedFar` by defaulting it in `defaultProps` here:
https://github.com/OfficeDev/office-ui-fabric-react/blob/79af8c83b984d3dd0a53470dcfbb058fa85f6567/packages/office-ui-fabric-react/src/components/Panel/Panel.base.tsx#L30-L36

The prop itself is optional so that is why I needed the null assertion here. I guess I can have the same default when deconstructing the `props` above.

Reply:
Removed the null assertion.